tests/unit/framework/db/QueryBuilderTest.php

```diff
@@ -228,6 +222,26 @@ public function conditionProvider()
 
         ];
 
+        switch ($this->driverName) {
+            case 'mssql':
+            case 'sqlite':
+                $conditions = array_merge($conditions, [
+                    [ ['in', ['id', 'name'], [['id' => 1, 'name' => 'foo'], ['id' => 2, 'name' => 'bar']]], '((`id` = :qp0 AND `name` = :qp1) OR (`id` = :qp2 AND `name` = :qp3))', [':qp0' => 1, ':qp1' => 'foo', ':qp2' => 2, ':qp3' => 'bar'] ],
+                    [ ['not in', ['id', 'name'], [['id' => 1, 'name' => 'foo'], ['id' => 2, 'name' => 'bar']]], '((`id` != :qp0 OR `name` != :qp1) AND (`id` != :qp2 OR `name` != :qp3))', [':qp0' => 1, ':qp1' => 'foo', ':qp2' => 2, ':qp3' => 'bar'] ],
+                    //[ ['in', ['id', 'name'], (new Query())->select(['id', 'name'])->from('users')->where(['active' => 1])], 'EXISTS (SELECT 1 FROM (SELECT `id`, `name` FROM `users` WHERE `active`=:qp0) AS a WHERE a.`id` = `id AND a.`name` = `name`)', [':qp0' => 1] ],
+                    //[ ['not in', ['id', 'name'], (new Query())->select(['id', 'name'])->from('users')->where(['active' => 1])], 'NOT EXISTS (SELECT 1 FROM (SELECT `id`, `name` FROM `users` WHERE `active`=:qp0) AS a WHERE a.`id` = `id` AND a.`name = `name`)', [':qp0' => 1] ],
```

Comment:
Creating a query like this would work:
~~~sql
EXISTS (
  SELECT 1
  FROM (
    SELECT c1, c2 FROM t1
  ) AS a
  WHERE a.c1 = top.c1 AND a.c2 = top.c2
)
~~~
There is a problem however how to extract the `top` alias name from the main query. We could try to parse the `from` property but there could be more than one table name there and there is no way to tell which one is the main one.
We could ignore that fact and just assume that columns passed in the condition should be aliased, but if they are not this fails silently, that is invalid results are produced and it will be hard to debug.

In the code above the `id` and `name` columns are not aliased and actually there isn't even a `from` value.